packages/amplify-category-api/src/provider-utils/awscloudformation/cdk-stack-builder/apigw-stack-builder.ts

```diff
@@ -64,28 +79,41 @@ export class AmplifyApigwResourceStack extends cdk.Stack implements AmplifyApigw
    * @param logicalId
    */
   addCfnResource(props: cdk.CfnResourceProps, logicalId: string): void {
+    if (this._seenLogicalIds.has(logicalId)) {
+      throw new Error(`logical id "${logicalId}" already exists`);
+    }
     new cdk.CfnResource(this, logicalId, props);
+    this._seenLogicalIds.add(logicalId);
   }
 
   /**
    *
    * @param props
    * @param logicalId
    */
-  addLambdaPermissionCfnResource(props: lambda.CfnPermissionProps, logicalId: string): void {
+  addCfnLambdaPermissionResource(props: lambda.CfnPermissionProps, logicalId: string): void {
+    if (this._seenLogicalIds.has(logicalId)) {
+      throw new Error(`logical id "${logicalId}" already exists`);
+    }
     new lambda.CfnPermission(this, logicalId, props);
+    this._seenLogicalIds.add(logicalId);
   }
 
   /**
    *
    * @param props
    * @param logicalId
+   * @param value optional value which will be stored in parameters.json
    */
-  addCfnParameter(props: cdk.CfnParameterProps, logicalId: string): void {
+  addCfnParameter(props: cdk.CfnParameterProps, logicalId: string, value?: string | $TSObject): void {
     if (this._cfnParameterMap.has(logicalId)) {
       throw new Error(`logical id "${logicalId}" already exists`);
     }
     this._cfnParameterMap.set(logicalId, new cdk.CfnParameter(this, logicalId, props));
+    this._seenLogicalIds.add(logicalId);
+    if (value !== undefined) {
```

Comment:
please thrown an exception if value is undefined.

Reply:
value is an optional parameter